crates/chainio/clients/elcontracts/src/writer.rs

```diff
@@ -256,4 +269,57 @@ mod tests {
             .await
             .unwrap());
     }
+
+    #[tokio::test]
+    async fn test_chain_writer() {
+        let (el_chain_reader, delegation_manager_address) = setup_el_chain_reader().await;
+        let http_endpoint = "http://localhost:8545".to_string();
+        let signer = "59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b78690d".to_string(); // TODO: retrieve this from anvil
+
+        let strategy_manager = get_strategy_manager_address().await;
+
+        let el_chain_writer = ELChainWriter::new(
+            delegation_manager_address,
+            strategy_manager,
+            el_chain_reader,
+            http_endpoint,
+            signer,
+        );
+
+        // define an operator
+        let wallet = PrivateKeySigner::from_str(
+            "bead471191bea97fc3aeac36c9d74c895e8a6242602e144e43152f96219e96e8",
+        )
+        .expect("no key");
+
+        let operator = Operator::new(
+            wallet.address(),
+            wallet.address(),
+            wallet.address(),
+            3,
+            Some("eigensdk-rs".to_string()),
+        );
+
+        let tx_hash = el_chain_writer
+            .register_as_operator(operator)
+            .await
+            .unwrap();
+
+        // this sleep is needed so that we wait for the tx to be processed
+        tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
+        let receipt = ANVIL_RPC_URL
+            .get_transaction_receipt(tx_hash)
+            .await
+            .unwrap();
+        assert!(receipt.unwrap().status());
+
+        // FIX this:
+        //     Transaction: 0xb497b163bef1f1f23dcc6bdc1cbaae653cb438dd3244940016c07a62642a8591
+        //     Gas used: 32928
+        //     Error: reverted with: revert: DelegationManager.registerAsOperator: caller is already actively delegated
```

Comment:
@pablodeymo 
this is due to this
https://github.com/Layr-Labs/eigensdk-rs/blob/749f634d192cbeea014b29d7a1040b7abbd3f660/crates/contracts/script/RegisterOperatorsWithEigenlayer.s.sol#L63

We register some anvil keys in scripts here already on eigen layer.

Reply:
this was already fixed, thanks 👍